core/encoding/json/marshal.odin

```diff
@@ -404,7 +407,16 @@ marshal_to_writer :: proc(w: io.Writer, v: any, opt: ^Marshal_Options) -> (err:
 		}
 
 	case runtime.Type_Info_Enum:
-		return marshal_to_writer(w, any{v.data, info.base.id}, opt)
+		if !opt.use_enum_value_names || len(info.names) == 0 {
+			return marshal_to_writer(w, any{v.data, info.base.id}, opt)
+		} else {
+			name, found := reflect.enum_name_from_value_any(v)
+			if found {
+				return marshal_to_writer(w, name, opt)
+			} else {
+				panic("Unable to find value in enum's values")
```

Comment:
Should this panic or return an error? I'm not sure myself.

Reply:
I don't really know either. I would normally log it & return an error. But I was going off the panics in the Union & Bit Set areas here. Where they panic if something's wrong with the type.

Reply:
That's a good point.
I like your suggestion layton. I'll add a fallback.